File: Preparation_Python/2025/DecimaltoBinary.py
def DecimaltoBinary(n):
    binary = ""
    temp = n
    while(temp>0):
        remainder = temp%2
        binary += str(remainder)
        temp //= 2
    return binary[::-1]
n = int(input())
print(DecimaltoBinary(n))

# Input: n = 12
# Output: "1100"

# Input: n = 33
# Output: "100001"

File: Practice/src/sep/DecimalToBinary.java
package sep;

import java.util.Scanner;

public class DecimalToBinary {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		
		StringBuilder sb = new StringBuilder();
		
		int temp = n;
		
		while(temp > 0) {
			int rem = temp % 2;
			sb.append(rem);
			temp = temp / 2;
		}
		sb.reverse();
        System.out.println("The binary equivalent of " + n + " is " + sb + ".");

	}

}
